Lib/test/string_tests.py

```diff
@@ -295,6 +295,32 @@ def test_index(self):
         else:
             self.checkraises(TypeError, 'hello', 'index', 42)
 
+        # For a variety of combinations,
+        #    verify that str.index() matches __contains__
+        #    and that the found substring is really at that location
+        charset = ['', 'a', 'b', 'c']
+        digits = 5
+        base = len(charset)
+        teststrings = set()
+        for i in range(base ** digits):
+            entry = []
+            for j in range(digits):
+                i, m = divmod(i, base)
+                entry.append(charset[m])
+            teststrings.add(''.join(entry))
+        teststrings = [self.fixtype(ts) for ts in teststrings]
+        for i in teststrings:
+            for j in teststrings:
+                try:
+                    loc = i.index(j)
+                except ValueError:
+                    loc = -1
```

Comment:
The test will pass if `index()` returns -1 (it should not). You should separate these cases.

Write different assertions for the case when `index()` returns and when it fails.
```py
                try:
                    loc = i.index(j)
                except ValueError:
                    ...
                else:
                    ...
```

Reply:
Yes, but the test does not catch this.